### PR TITLE
Scopes: Fix rendering of the scopes list for large chunk of items

### PR DESCRIPTION
**Pull request by bfmatei**
**What is this feature?**

This PR brings virtual scrollbars to the scopes list.

**Why do we need this feature?**

In order to prevent the apply/cancel buttons to overflow the screen. Check the demo:

https://github.com/user-attachments/assets/aa3230d8-cdd9-4fb8-b6b7-12803c884335

**Who is this feature for?**

Users of scopes

**Which issue(s) does this PR fix?**:

-

**Special notes for your reviewer:**

You can test this more easily if you use the following `fetchNodes` adaptation in `public/app/features/scopes/internal/api.ts`:

```
export async function fetchNodesNew(parent: string, query: string): Promise<NodesMap> {
  let a = 0;
  return (await fetchScopeNodes(parent, query)).reduce<NodesMap>((acc, { metadata: { name }, spec }) => {
    acc[name] = {
      name,
      ...spec,
      isExpandable: spec.nodeType === 'container',
      isSelectable: spec.linkType === 'scope',
      isExpanded: false,
      query: '',
      reason: NodeReason.Result,
      nodes: {},
    };

    for (let i = 0; i < (a === 0 ? 100 : 5); i++) {
      name = `${name}${Math.random()}`;
      acc[name] = {
        name,
        ...spec,
        linkId: spec.linkId ? `${spec.linkId}-${i}` : undefined,
        isExpandable: spec.nodeType === 'container',
        isSelectable: spec.linkType === 'scope',
        isExpanded: false,
        query: '',
        reason: NodeReason.Result,
        nodes: {},
      };
    }
    a++;
    return acc;
  }, {});
}

export async function fetchNodes(parent: string, query: string): Promise<NodesMap> {
  if (parent === 'cloud-applications' && query !== '' && query !== 'Tempo') {
    return fetchNodesNew(parent, query);
  }

  return fetchNodesOrig(parent, query);
}
```

Do update the code for your needs.

Please check that:
- [x] It works as expected from a user's perspective.
- [x] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/contribute/release-notes/#how-to-determine-if-content-belongs-in-whats-new), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/contribute/release-notes/) doc.

### CODE REVIEW

**Review by harisrozajac — APPROVED** [review]
Looks great. Probably not related to this PR, but the focus border around the initial search bar and radio buttons (doesn't happen with checkboxes) is cut off a bit on the left:
Screenshot 2024-10-24 at 4 59 05 PM
Screenshot 2024-10-24 at 4 59 29 PM

**Review by mdvictor — APPROVED** [review]
Works great and looks great. Tested even having both leaf scopes list and results scope list with many values to replicate having scrollbar in both divs, works like a charm.

@harisrozajac is right though, the focus on input and radios is cut left-side. 

This doesn't repro on main so should be addressed by this PR

File: public/app/features/scopes/internal/ScopesSelectorScene.tsx [hunk 1]
```diff
@@ -425,6 +425,8 @@ const getStyles = (theme: GrafanaTheme2, menuDockedAndOpen: boolean) => {
       flexDirection: 'column',
       maxHeight: '100%',
       overflowY: 'hidden',
+      // Fix for top level search outline overflow due to scrollbars
+      paddingLeft: theme.spacing(0.5),
```
Comment (mdvictor) [inline public/app/features/scopes/internal/ScopesSelectorScene.tsx:429]: Nice 💯

**Comment by bfmatei** [comment]
@harisrozajac @mdvictor thank you so much. Fixed the outlines for both the search bars and for the checkboxes/radios.

**Pull request merged**